src/Clockify/ClockifySetupDialog.cs

```diff
@@ -62,15 +66,23 @@ public ClockifySetupDialog(UserState userState, IClockifyService clockifyService
             CancellationToken cancellationToken)
         {
             string? token = promptContext.Recognized.Value;
-            
+
+            if (string.IsNullOrWhiteSpace(token) || token.Length > 100)
+            {
+                return false;
+            }
+
             try
             {
                 var userProfile = await _userState.CreateProperty<UserProfile>("UserProfile")
                     .GetAsync(promptContext.Context, () => new UserProfile(), cancellationToken);
 
                 string? userId = _clockifyService.GetCurrentUserAsync(token).Result.Id;
-
-                userProfile.ClockifyToken = token;
+                
+                var tokenData = await _tokenRepository.WriteAsync(token, userProfile.ClockifyTokenId);
+                userProfile.ClockifyToken = tokenData.Value;
```

Comment:
why is this here? Do we really need a stepped rollout? 
If it's properly tested I think we can go all-in with the KV without having a 2-way phase

Or am I missing something?

Reply:
We can, but each user should re-enter the token, however, in this case line 83 can be erased